src/main/kotlin/lotto/domain/LottoNumbers.kt

```diff
@@ -0,0 +1,53 @@
+package lotto.domain
+
+import lotto.argumentError
+
+class LottoNumbers(
+    private val numbers: List<LottoNumber>
+) {
+    init {
+        check(numbers.size == LOTTO_NUMBER_SET) {
+            argumentError("로또는 중복 없는 6자리 번호여야합니다.")
+        }
+    }
+
+    fun matches(inputNumbers: LottoNumbers): Int {
+        return numbers.count { number ->
+            inputNumbers.numbers.contains(number)
+        }
+    }
+
+    override fun toString(): String {
+        return numbers.joinToString(", ") { number -> number.toString() }
+    }
+
+    companion object {
+        private const val LOTTO_NUMBER_SET = 6
+        private val LOTTO_NUMBERS = (1..45)
+
+        fun from(): LottoNumbers {
+            val selected = LOTTO_NUMBERS.shuffled()
+                .take(6)
+                .sorted()
+                .map { number ->
+                    LottoNumber(number)
+                }
+
+            return LottoNumbers(
+                selected
+            )
+        }
```

Comment:
```suggestion
	fun from(): LottoNumbers {
		return LottoNumbers(LOTTO_NUMBERS.shuffled()
			.take(6)
			.sorted()
			.map { number ->
				LottoNumber(number)
			}
		)
	}
```

한번에 초기화 해줄 수 있습니다 :)

Reply:
step3에 적용하였습니다